astropy/convolution/tests/test_kernel_class.py

```diff
@@ -292,7 +292,7 @@ def test_custom_1D_kernel_list(self):
         Check if CustomKernel works with lists.
         """
         custom = CustomKernel([1, 1, 1, 1, 1])
-        assert custom.is_bool == True
```

Comment:
I am wary of all removal of `== True` in this PR (across multiple sub-packages). Maybe it is safer to do `is True` unless you are very sure removing it will not change the context of the comparison.